libs/core-components/src/lib/TextCopyField/TextCopyField.tsx

```diff
@@ -1,66 +1,77 @@
 import React, { useCallback, useState } from 'react';
-import { CopyToClipboard } from 'react-copy-to-clipboard';
-import PropTypes from 'prop-types';
 
 import Button from '../Button';
 
 import styles from './TextCopyField.module.css';
 
-const TextCopyField = ({ value, placeholder, buttonWrapper }) => {
+type TextCopyFieldProps = {
+  value: string;
+  placeholder?: string;
+  className?: string;
+  buttonClassName?: string;
+  ButtonWrapper?: React.FC<React.PropsWithChildren>;
+};
+
+const TextCopyField: React.FC<TextCopyFieldProps> = ({
+  value,
+  placeholder,
+  className,
+  buttonClassName,
+  ButtonWrapper,
+}) => {
+  /* WARNING: Must match CSS `--transition-duration` */
   const transitionDuration = 0.15; // second(s)
   const stateDuration = 1; // second(s)
   const stateTimeout = transitionDuration + stateDuration; // second(s)
 
   const [isCopied, setIsCopied] = useState(false);
 
   const onCopy = useCallback(() => {
+    navigator.clipboard.writeText(value);
     setIsCopied(true);
 
     const timeout = setTimeout(() => {
       setIsCopied(false);
       clearTimeout(timeout);
     }, stateTimeout * 1000);
-  }, [setIsCopied, stateTimeout]);
+  }, [value, setIsCopied, stateTimeout]);
   const isEmpty = !value || value.length === 0;
-  const onChange = (event) => {
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     // Swallow keyboard events on the Input control, but
     // still allow selecting the text. readOnly property of
     // Input is not adequate for this purpose because it
     // prevents text selection
     event.preventDefault();
   };
 
-  const ButtonWrapper = buttonWrapper;
-  const CopyButton = (
-    <CopyToClipboard text={value}>
+  const CopyButton = () => {
+    return (
       <Button
         className={`${styles['copy-button']} ${
           isCopied ? styles['is-copied'] : ''
-        }`}
-        // RFE: Avoid manual JS ↔ CSS sync of transition duration by using:
-        //      - `data-attribute` and `attr()` (pending browser support)
-        //      - PostCSS and JSON variables (pending greater need for this)
-        style={{ '--transition-duration': `${transitionDuration}s` }}
+        } ${buttonClassName}`}
+        size="small"
         onClick={onCopy}
         disabled={isEmpty}
         iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
-        type="button"
       >
         Copy
       </Button>
-    </CopyToClipboard>
-  );
-  const CopyField = (
-    <input
-      type="text"
-      onChange={onChange}
-      value={value}
-      className={`form-control ${styles.input}`}
-      placeholder={placeholder}
-      data-testid="input"
-      readOnly
-    />
-  );
+    );
+  };
+  const CopyField = () => {
```

Comment:
Why is the CopyField an input element? There's no case where someone would want to change the value on the screen.

Reply:
Uncertain.[^1] But I agree.

[^1]: I converted it from J.Y. who got it from his code in Core-Portal.